.devcontainer/devcontainer.json

```diff
@@ -1,22 +1,34 @@
-// For format details, see https://aka.ms/devcontainer.json. For config options, see the
-// README at: https://github.com/devcontainers/templates/tree/main/src/miniconda
-{
-	"name": "DSToolKit",
-	"context": ".",
-	"dockerFile": "Dockerfile",
-
-	// Set *default* container specific settings.json values on container create.
-	"settings": {
-		"terminal.integrated.shell.linux": "/bin/bash"
-	},
-
-	// Add the IDs of extensions you want installed when the container is created.
-	"extensions": [
-		"ms-python.python",
-		"ms-toolsai.vscode-ai",
-		"ms-toolsai.jupyter",
-		"redhat.vscode-yaml"
-	],
-
-	"runArgs": ["-v", "/var/run/docker.sock:/var/run/docker.sock"]
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/miniconda
+{
+	"name": "DSToolKit",
+	"context": ".",
+	"dockerFile": "Dockerfile",
+
+	"customizations": {
+		"vscode": {
+			"settings": {
+				"terminal.integrated.shell.linux": "/bin/bash",
+				"[python]": {
+					"editor.defaultFormatter": "ms-python.black-formatter",
+					"editor.formatOnSave": true,
+					"editor.codeActionsOnSave": {
+						"source.organizeImports": "explicit"
+					}
+				},
+				"isort.args":["--profile", "black"]
+			},
+			"extensions": [
+				"leojhonsong.python-extension-pack",
+				"njqdev.vscode-python-typehint",
+				"formulahendry.code-runner",
+				"redhat.vscode-yaml",
+				"streetsidesoftware.code-spell-checker",
+				"ms-python.black-formatter",
+				"ms-python.isort"
+			]
+		}
+	},
+	"runArgs": ["-v", "/var/run/docker.sock:/var/run/docker.sock"],
+	"postStartCommand": "pre-commit install"

```

Comment:
I think it's recommended to not have pre-commit installed by default and instead instruct the developers on how to install it manually.

[Playbook reference](https://microsoft.github.io/code-with-engineering-playbook/code-reviews/recipes/python/#pre-commit-hooks)

![image](https://github.com/user-attachments/assets/c6b1608b-89c5-4191-9d75-3b3ba3b2231a)